libs/langchain/tests/unit_tests/chat_models/test_huggingface_llama2.py

```diff
@@ -0,0 +1,37 @@
+"""Test Hugging Face Llama-2 Chat model."""
+
+from langchain.chat_models.huggingface_llama2 import ChatLlama2Hf
+from langchain.schema.messages import (
+    AIMessage,
+    HumanMessage,
+    SystemMessage,
+)
+
+
+def test_format_messages_as_text_with_system() -> None:
+    messages = [
+        SystemMessage(content="System Prompt."),
+        HumanMessage(content="Human Message."),
+        AIMessage(content="AI response."),
+        HumanMessage(content="Second Human Message."),
+        AIMessage(content="AI response."),
+    ]
+
+    ground_truth = "<s>[INST] <<SYS>>\nSystem Prompt.\n<</SYS>>\n\nHuman Message. [/INST] AI response. </s><s>[INST] Second Human Message. [/INST] AI response. </s><s>[INST] "  # noqa: E501
```

Comment:
in python "foo" "bar" is merged to "foobar", so you can likely keep it shorter by string split